lib/ansible/modules/cloud/amazon/ec2_win_password.py

```diff
@@ -151,16 +166,22 @@ def main():
     if wait and datetime.datetime.now() >= end:
         module.fail_json(msg="wait for password timeout after %d seconds" % wait_timeout)
 
-    try:
-        f = open(key_file, 'rb')
-    except IOError as e:
-        module.fail_json(msg="I/O error (%d) opening key file: %s" % (e.errno, e.strerror))
-    else:
+    if key_file is not None and key_data is None:
+        try:
+            f = open(key_file, 'rb')
+        except IOError as e:
+            module.fail_json(msg="I/O error (%d) opening key file: %s" % (e.errno, e.strerror))
+        else:
+            try:
+                with f:
```

Comment:
You could also unnest this a bit and do
```
try:
    with open(key_file, 'rb') as f:
        key = load_pem_private_key(f.read(), b_key_passphrase, default_backend())
except IOError as e:
    # Handle bad files
except (ValueError, TypeError) as e:
    # Handle issues loading key
```

Reply:
Thanks for the pointer, I've added another commit.